addons/website_sale_delivery/static/src/js/website_sale_delivery.js

```diff
@@ -35,24 +39,207 @@ publicWidget.registry.websiteSaleDelivery = publicWidget.Widget.extend({
+            error_message.classList.add("text-error");
+            error_message.innerText = data.error ? data.error : "No available Pick-Up Locations";
+            modal.appendChild(error_message);
+            return;
+        }
+        
+        var list_to_render = delivery_type + "_pickup_location_list";
+        var data_to_render = {partner_address: data.partner_address};
+        data_to_render[delivery_type + "_pickup_locations"] = data.close_locations;
+        $(qweb.render(list_to_render, data_to_render)).appendTo($(modal));
+        this._displayCarrierDropper(ev);  
+    },
+    /**
+     * @private
+     * @param {Object} obj // can be either an event or an objet returned by the controller
+     */
+    _displayCarrierDropper: async function (obj) {
+        let current_id = obj.carrier_id
+        const show_locations = document
+            .getElementById("delivery_carrier")
+            .getElementsByClassName("o_show_pickup_locations");
+
+        // We check if it's an event
+        if('currentTarget' in obj){
+            const is_full_page_event = !obj.currentTarget.closest(".o_delivery_carrier_select")
+            if(is_full_page_event)return;
+            for (const show_loc of show_locations){
+                this._specificDropperDisplay(show_loc);
+            }
+        }
+        else{
+            for (const show_loc of show_locations){
+                const current_carrier_id = show_loc.closest("li").getElementsByTagName("input")[0].value;
+                if (current_carrier_id == current_id) {
+                    this._specificDropperDisplay(show_loc);
+                    break;
+                }
+            }
+        }
+    },
+    /**
+     * @private
+     * @param {Object} doc_carrier //carrier element from document
+     */
+    _specificDropperDisplay: function (doc_carrier) {
+        if(!doc_carrier)
+            return;
+        if(!doc_carrier.closest("li").getElementsByTagName("input")[0].attributes.getNamedItem("delivery_type"))
+            return;
+        let current_carrier_type = doc_carrier.closest("li").getElementsByTagName("input")[0].attributes.getNamedItem("delivery_type").nodeValue;
+        const location_still_loading = document.getElementById("#"+current_carrier_type+"_addresses")
+        if(location_still_loading)return
+        while (doc_carrier.firstChild) {
+            doc_carrier.lastChild.remove();
+        }
+        const current_carrier_checked = doc_carrier.closest("li").getElementsByTagName("input")[0].checked;
+        const span = document.createElement("em");
+        const is_pick_up_location_list = doc_carrier.nextElementSibling.childNodes.length > 0
+        if(current_carrier_checked){
+            const chevron_down = document.createElement("i");
+            if(is_pick_up_location_list)
+                chevron_down.classList.add("fa", "fa-angle-up");
+            else
+                chevron_down.classList.add("fa", "fa-angle-down");
```

Comment:
Always use braces on new branches/blocks, it makes the code way less error prone.
```suggestion
            if(is_pick_up_location_list) {
                chevron_down.classList.add("fa", "fa-angle-up");
            } else {
                chevron_down.classList.add("fa", "fa-angle-down");
            }
```